### PR TITLE
Feature/12527 fss UI connect search filter panel with get batch attributes endpoint

### PR DESCRIPTION
**Pull request by gaurav15233**
Hi All, Please review and merge changes related to connect search filter panel with get batch attributes endpoint for FSS UI in main branch.

### CODE REVIEW

**Review by jimjamjahaa — COMMENTED** [review]
lgtm but the tests are failing atm

File: src/app/features/fss-search/fss-search.component.ts [hunk 1]
```diff
@@ -118,6 +126,30 @@ export class FssSearchComponent implements OnInit {
...
+          items : this.getAttributesValues(batchAttributesResult[i].values),
+          expanded : true
+        });
+      }
+    }
+  }
+
+  getAttributesValues(batch:Array<any> = []) {
+    var batchAttributesValues: FilterItem[] = [];
+    for (var i = 0; i < batch.length; i++) {
```
Comment (jimjamjahaa) [inline src/app/features/fss-search/fss-search.component.ts:144]: another var that should be let

**Review by jimjamjahaa — CHANGES_REQUESTED** [review]
i appologise for not picking these things up first time round but there are some things to look at

this PR seems to have similar code to PR 91 so i hope 2 people aren't trying to fix the same thing :)

File: Tests/UnitTests/fss-search.component.spec.ts [hunk 2]
```diff
@@ -52,8 +53,92 @@ describe('FssSearchComponent', () => {
     const app = fixture.componentInstance;
     expect(app).toBeTruthy();
   });
+
+  it('should return filter groups when search result data is provided', () => {
+    var inputSearchResultData = <JSON>inputSearchResultMockData;
```
[the remark this answers is not in the document]
Comment (gaurav15233) [reply Tests/UnitTests/fss-search.component.spec.ts]: updated as suggested. Thanks

File: Tests/UnitTests/fss-search.component.spec.ts [hunk 3]
```diff
@@ -52,8 +53,92 @@ describe('FssSearchComponent', () => {
...
+    var expectedResultBatchAttributes = <JSON>attributeSearchFilterMockData;
+    component.transformSearchAttributesToFilter(inputSearchResultData);
+    expect(component.filterGroups.length).toEqual(4);
+    expect(component.filterGroups).toEqual(expectedResultBatchAttributes);
+  });
 });
 
+
+export var inputSearchResultMockData: any =
```
[the remark this answers is not in the document]
Comment (gaurav15233) [reply Tests/UnitTests/fss-search.component.spec.ts]: updated as suggested. Thanks

File: src/app/core/services/file-share-api.service.ts [hunk 4]
```diff
@@ -24,6 +24,10 @@ export class FileShareApiService {
         }
     }
 
+    getAttributeSearchResult(payload: string): Observable<any> {
+                return this.http.get(this.baseUrl + "/attributes/search?$filter=" + encodeURIComponent(payload));
```
[the remark this answers is not in the document]
Comment (gaurav15233) [reply src/app/core/services/file-share-api.service.ts]: updated as suggested. Thanks

File: src/app/features/fss-search/fss-search.component.ts [hunk 5]
```diff
@@ -118,6 +126,30 @@ export class FssSearchComponent implements OnInit {
     this.analyticsService.getSimplifiedSearchResult();
   }
 
+  transformSearchAttributesToFilter(batchAttributesResult :any) {
+    this.filterGroups = [];
+    if (batchAttributesResult.length > 0) {
+      for (var i = 0; i < batchAttributesResult.length; i++) {
```
[the remark this answers is not in the document]
Comment (gaurav15233) [reply src/app/features/fss-search/fss-search.component.ts]: Similar comments we resolved in PR 91

File: src/app/features/fss-search/fss-search.component.ts (hunk 1 above)
Comment (gaurav15233) [reply src/app/features/fss-search/fss-search.component.ts]: updated as suggested. Thanks

**Review by bhalcha13813 — APPROVED** [review]
LGTM

**Review by jimjamjahaa — COMMENTED** [review]
LGTM but there are merge conflicts

**Comment by gaurav15233** [comment]
Hi Richard, I have resolved the merge conflicts.
Thank you.

**Pull request merged**